### PR TITLE
Add more APIs to String and Char types with supported tests

### PR DESCRIPTION
**Pull request by tarekgh**

### CODE REVIEW

**Comment by tarekgh** [comment]
CC @weshaggard @danmosemsft @stephentoub

File: src/System.Runtime/tests/System/StringTests.netstandard1.7.cs [hunk 1]
```diff
@@ -118,5 +130,99 @@ public static void StartEndNegativeTest()
...
+
+            Assert.False(s.IsNormalized(), "String should be not normalized when checking with the default which same as FormC");
+            Assert.False(s.IsNormalized(NormalizationForm.FormC), "String should be not normalized when checking with FormC");
+            Assert.False(s.IsNormalized(NormalizationForm.FormD), "String should be not normalized when checking with FormD");
+            Assert.False(s.IsNormalized(NormalizationForm.FormKC), "String should be not normalized when checking with FormKC");
+            Assert.False(s.IsNormalized(NormalizationForm.FormKD), "String should be not normalized when checking with FormKD");
+
+            string normalized = s.Normalize(); // FormC
+            Assert.True(normalized.IsNormalized(), "Expected to have the normalized string with default form FormC");
+            Assert.True(normalized.IsNormalized(NormalizationForm.FormC), "Expected to hav eth enormalized string with FormC");
```
Comment (danmosemsft) [inline src/System.Runtime/tests/System/StringTests.netstandard1.7.cs]: typo

Comment (tarekgh) [inline src/System.Runtime/tests/System/StringTests.netstandard1.7.cs]: will fix that.

File: src/System.Runtime/tests/System/CharTests.netstandard1.7.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+using System.Globalization;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.Tests
+{
+    public static class CharMiscTests
+    {
+        private static UnicodeCategory[] s_categoryForLatin1 =
```
Comment (justinvp) [inline src/System.Runtime/tests/System/CharTests.netstandard1.7.cs]: Nit: readonly

Comment (tarekgh) [inline src/System.Runtime/tests/System/CharTests.netstandard1.7.cs]: fixed it. thanks.

**Pull request merged**